### ISSUE TITLE
Passing Function as Child

### ISSUE THREAD

**Issue by ElGrecode** [ethanselzer/react-hover-observer#2]
Hey @ethanselzer. Thanks for building this package, it's been very useful for me and I'm very grateful. One of the things I'd like to do to extend it is allow the passing of a function as a HoverObservers's children prop. This would make it a lot more flexible. Currently, what I have to do is explicitly make a new component(s) so the component can be decorated with the prop `isHovering`.

For this to make a little sense I'd like to show you an example.
```
<ReactHoverObserver hoverOffDelayInMs={100}>
     <Tabs
         isActive={}
      />
</ReactHoverObserver>
```

In the example above, Tabs is already a very simple component that has a semantic `isActive` prop. In this use case, I'd like for my tab to be active when I am hovering... In other tabs on my site I'd like them to be active only when clicking. Clearly two different use cases.

To pass `isHovering` to my `<Tabs />` component, currently I would have to make another component because of how `react-hover-observer` decorates a component with `isHovering`.

```
<ReactHoverObserver hoverOffDelayInMs={100}>
     <SpecificHoveringTabs/>
</ReactHoverObserver>
```

```
// SpecificHoveringTabs.js
...
return (
  <Tabs isActive={this.props.isHovering}
)
```

What I'd like to do is be able to specify a function to be passed as a child. So that I can be more flexible with the children to ReactHoverObserver. For example...

```
<ReactHoverObserver hoverOffDelayInMs={100}>
  {(props) => {
    const {isHovering} = this.props;
    return (
      <Tabs
         isActive={isHovering}
      />
    )
  }
</ReactHoverObserver>
```

What do you think about this idea?

**Comment by ethanselzer** [comment]
@ElGrecode - Thanks for opening this issue. I really appreciate the detail you provided! I think the pattern you suggest is great. I would like to continue supporting the "decorator" style pattern also, so I'm thinking of inspecting the type of `children` and adapting accordingly. I'm wondering if you know of any problems with this approach.

**Comment by ElGrecode** [comment]
Nice! I think that works great as a solution. It makes it very adaptable to many use cases.

**Comment by ethanselzer** [comment]
Okay great. I'll get to work on it!

**Comment by ethanselzer** [comment]
@ElGrecode - Okay all set. Support for function as a child component has landed in [v2.1.0](https://www.npmjs.com/package/react-hover-observer). Please have a look and let me know if you discover any issues. Thanks again for suggesting the feature. I think it is a big improvement! If you find react-hover-observer useful, please consider starring it on Github. Starring helps to grow the project, which benefits everyone.

**Comment by ethanselzer** [comment]
Closing as fixed by [v2.1.0](https://github.com/ethanselzer/react-hover-observer/releases/tag/v2.1.0).
